UnityProject/Assets/Scripts/Items/Tools/Spray.cs

```diff
@@ -0,0 +1,178 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Spray
+{
+    /// <summary>
+    /// Used to call a spray coming from the player, where position is the target, bottle is the ReagentContainer of the item,
+    /// spray is the prefab to spray, and customCol and col are for if you want the spray to have custom colours.
+    /// </summary>
+    /// <param name="bottle"></param>
+    /// <param name="position"></param>
+    /// <param name="spray"></param>
+    /// <param name="customCol"></param>
+    /// <param name="col"></param>
+    /// <returns></returns>
+    public static IEnumerator TriggerSpray(ReagentContainer bottle, Vector3 position, GameObject spray, bool customCol = false, Color col = new Color())
```

Comment:
It's ok to do it this way (assuming it works even over the network, haven't tested it yet), but just wanted to point out a more "Unity-like" way...

With this approach you have all of the logic for controlling the sprays living in this Util class. The spray prefab itself isn't doing anything except having a sprite. It's being controlled externally by this method.

Another approach would be to attach a new component (for example lets call it Spray) to the spray prefab so that each spray controlled itself. That way the game object controls itself. When you want to spray, you just spawn an instance of the prefab and set its direction / chemical / whatever needs to be set, then let it go. Once a spray has been created, it should handle its own logic.

That would also avoid you needing to use a Coroutine to handle the spray logic. You could allow the spray to move itself with its own Spray component, such as in the Update method.

It's usually a good practice to use Unity components for things.